spec/lib/jit_preloader/active_record/base_spec.rb

```diff
@@ -44,6 +44,14 @@ def call(contact)
       end.to make_database_queries(count: 1)
     end
 
+    context "with newly created records" do
+      it "ignores new records in scope" do
+        contact = Contact.first
+        new_address = contact.addresses.create!(street: "123 Friendly St.", country: canada)
```

Comment:
I might be misunderstanding the test here, but based on my reading `call` is passing in the country of `:usa` - does this object need to have `country: :usa` too? I _think_ that right now because country is canada it will be ignored but moreso because of the implementation of `call` than because of the `target` change

Reply:
Because it has country of canada it *should* be ignored.

Call preloads addresses in the USA.

However, due to a bug, newly created records - `new_address`, in Canada - would *also* be included as we preload.

I'll be honest, this exposes some of the ugliness with `preload_scoped_relation` - we find an association, save it's state, trick rails into running preloading code with an altered scope, pull the loaded records of the association, then reset the association to its original state.  Seems vulnerable to change in Rails preloaders.

Reply:
Or - easy response - is the `not_to include` confusing?

Reply:
🤦 ooooh. I understand now. The `not_to` wasn't confusing but I did jump up to `call` and then seeing the query made me wonder. Not sure if there's a clearer way to communicate this one though